cli/src/config.rs

```diff
@@ -0,0 +1,66 @@
+use lazy_static::lazy_static;
+use log::info;
+use serde::Deserialize;
+use std::{char, path::Path, process::Command};
+
+const FILE_NAME: &str = ".squawk.toml";
+lazy_static! {
+    static ref STATIC_SEARCH_PATHS: Vec<String> =
+        vec![format!("./{}", FILE_NAME), format!("~/{}", FILE_NAME)];
+}
+
+#[derive(Debug, Default, Deserialize)]
+pub struct Config {
+    pub excluded_rules: Option<Vec<String>>,
+}
+
+impl Config {
+    pub fn parse(custom_path: Option<String>) -> Option<Self> {
+        let path = custom_path.or_else(Self::find_path);
+
+        if let Some(p) = path {
+            info!("config file path: {}", &p);
+            if Path::new(&p).exists() {
+                if let Ok(file_content) = std::fs::read_to_string(p) {
+                    return toml::from_str(&file_content).ok();
+                }
+            }
+        }
+
+        info!("no config file found");
+        None
+    }
+
+    fn find_path() -> Option<String> {
+        let static_path = STATIC_SEARCH_PATHS
+            .iter()
+            .find(|p| Path::new(p).exists())
+            .map(String::to_string);
+
+        // Config in git root takes priority after the local directory config
+        match (&static_path, get_git_root_config_path()) {
+            (Some(sp), Some(git_root_config_path)) => {
+                if sp == &STATIC_SEARCH_PATHS[0] {
+                    static_path
+                } else {
+                    Some(git_root_config_path)
+                }
+            }
+            (None, Some(git_root_config_path)) => Some(git_root_config_path),
+            _ => static_path,
+        }
+    }
+}
+
+fn get_git_root_config_path() -> Option<String> {
```

Comment:
I wonder if instead of calling out to git, we could instead navigate up the file tree until we find a `.squawk.toml`

Reply:
I'm not fond of going back until a `.squawk.toml` is found because the location of where it ends up it's not known to the user, and it might lead to using a config file that the user doesn't intend to which won't be easy to notice without using the `--verbose` flag.

If your concern is the behavior of the program for users without the `git` binary then I just tested that the `get_git_root_config_path` function just returns `None` in case it can't run the binary, which lets the program continue running without panic or crash.

Let me know if you have other concerns about calling `git`  🙂

Reply:
I think having the dependency on `git` itself for finding the config file isn't the best since not everyone uses git

Instead I think traversing up the path until we find the config would be better

Reply:
Okay, well if you prefer that approach I will implement it 👍

Reply:
Implemented 👉  96f8d29